File: app/src/main/java/com/github/mrvilkaman/presentationlayer/fragments/testfrags/FragBaseScreenFragment.java
package com.github.mrvilkaman.presentationlayer.fragments.testfrags;


import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.github.mrvilkaman.R;
import com.github.mrvilkaman.presentationlayer.fragments.core.BaseFragment;
import com.github.mrvilkaman.presentationlayer.resolution.toolbar.IToolbar;

import butterknife.BindView;
import butterknife.OnClick;

public abstract class FragBaseScreenFragment extends BaseFragment<FragBasePresenter> {

 	@BindView(R.id.frag_text)TextView numberText;

	@Override
	protected int getLayoutId() {
		return R.layout.layout_fragscreen_fragment;
	}

	@Override
	protected void onCreateView(View view, Bundle savedInstanceState) {
		IToolbar toolbar = getToolbar();
		toolbar.show();
		toolbar.showIcon(getIcon(),() -> showToast(R.string.app_name));
		String text = Integer.toString(getNumber());
		toolbar.setTitle(text);
		numberText.setText(text);
	}

	protected int getIcon() {
		return R.drawable.ic_home;
	}

	protected abstract int getNumber();

	protected abstract BaseFragment nextFragment();

	@OnClick(R.id.frag_stack)
	public void onClickStack() {
		getNavigation().showFragment(nextFragment());
	}

	@OnClick(R.id.frag_no_stack)
	public void onClickNoStack() {
		getNavigation().showFragmentWithoutBackStack(nextFragment());
	}

	@OnClick(R.id.frag_root)
	public void onClickRoot() {
		getNavigation().showRootFragment(Frag1ScreenFragment.open());
	}

}

File: kotlinapp/src/main/java/com/github/mrvilkaman/ui/screens/testfrags/FragBaseScreenFragment.kt
package com.github.mrvilkaman.ui.screens.testfrags


import android.os.Bundle
import android.view.View
import com.github.mrvilkaman.R
import com.github.mrvilkaman.presentationlayer.fragments.core.BaseFragment
import kotlinx.android.synthetic.main.layout_fragscreen_fragment.*
import ru.terrakok.cicerone.Router
import javax.inject.Inject

abstract class FragBaseScreenFragment : BaseFragment<FragBasePresenter>() {

    @Inject lateinit open var router: Router

    protected open val icon: Int
        get() = R.drawable.ic_home

    protected abstract val number: Int

    override fun getLayoutId(): Int = R.layout.layout_fragscreen_fragment

    override fun onCreateView(view: View, savedInstanceState: Bundle?) {
        val text = Integer.toString(number)
        val toolbar = getToolbar()
        if (toolbar != null) {
            toolbar.show()
            toolbar.showIcon(icon) {
                getUiResolver().showToast(R.string.app_name)
            }
            toolbar.setTitle(text)
        }

        frag_text.text = text

        frag_stack.setOnClickListener { router.navigateTo(nextScreenKey()) }
        frag_no_stack.setOnClickListener { router.replaceScreen(nextScreenKey()) }
        frag_root.setOnClickListener { router.newRootScreen(nextScreenKey()) }
    }


    protected abstract fun nextScreenKey(): String?

}